tests/zfs-tests/tests/functional/fault/auto_online_001_pos.ksh

```diff
@@ -0,0 +1,148 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+#
+# Copyright (c) 2016 by Intel Corporation. All rights reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+. $STF_SUITE/tests/functional/fault/fault.cfg
+
+#
+# DESCRIPTION:
+# Tesing auto-online FMA ZED logic.
+#
+# Method: Creates a pool, exports pool, offlines disk,
+# imports pool with missing disk, then onlines disk.
+# ZED polls for an event change for online disk to be automatically
+# added back to the pool.
+# Creates a raidz1 zpool using persistent disk path names
+# (ie not /dev/sdc). Tests import using pool guid and cache file.
+#
+# If loop devices are used, then a scsi_debug device is added to the pool.
+
+verify_runnable "both"
+
+if ! is_physical_device $DISKS; then
+	log_unsupported "Unsupported disks for this test."
+fi
+
+function cleanup
+{
+	#online last disk before fail
+	on_off_disk $offline_disk "online"
+	poolexists $pool && destroy_pool $pool
+}
+
+log_assert "Testing auto-online FMA ZED logic"
+
+log_onexit cleanup
+
+pool=auto-online_test
+fault_cachefile=$STF_SUITE/tests/functional/fault/zpool_fault.cache
+target=$pool
+
+if is_loop_device $disk1; then
+	SD=$($LSSCSI | $NAWK '/scsi_debug/ {print $6; exit}')
+	SDDEVICE=$($ECHO $SD | $NAWK -F / '{print $3}')
+	SDDEVICE_ID=$(get_persistent_disk_name $SDDEVICE)
+	autoonline_disks="$SDDEVICE"
+else
+	autoonline_disks="$disk1 $disk2 $disk3"
+fi
+
+# Clear disk labels
+for i in {0..2}
+do
+	log_must $ZPOOL labelclear -f /dev/disk/by-id/"${devs_id[i]}"
+done
+
+for import_option in "-c $fault_cachefile -a" "$target"
+do
+	if is_loop_device $disk1; then
+		#create a pool with one scsi_debug device and 3 loop devices
+		log_must $ZPOOL create -f $pool raidz1 $SDDEVICE_ID $disk1 \
+		    $disk2 $disk3
+	elif ( is_real_device $disk1 || is_mpath_device $disk1 ); then
+		log_must $ZPOOL create -f $pool raidz1 ${devs_id[0]} \
+		    ${devs_id[1]} ${devs_id[2]}
+	else
+		log_fail "Disks are not supported for this test"
+	fi
+
+	#add some data to the pool
+	log_must $TRUNCATE -s $FSIZE /$pool/data
```

Comment:
`$MKFILE $FSIZE $TESTPOOL/data ` might be a better way to do this since then you won't get a sparse file.